### DISCUSSION TITLE
Http server for discord bot.

### DISCUSSION THREAD

**Discussion by denisyakimov07** [Rapptz/discord.py#9751] [Q&A] [answered]
Hello, could you please recommend http server  to add into my discord bot to be able to trigger discord functions from custom webhooks.
I play around flask but i can call discord funtions from ather tread.

**Comment by No767** [comment]
@denisyakimov07 You would use AIOHTTP, but your case is different than most usecases. You want to trigger some Discord function based off when webhooks are emitted. This simply is not possible, as it's not possible to listen for webhooks with discord.py. Instead of relying on an weak link such as depending on an custom webhook event as this exposes the weak links in your chain, an alternative must be used. I would be interested in *what exactly* you are trying to do. In short, what is your goal with your feature?

**Comment by denisyakimov07** [comment]
@No767  My game server send logs to API, and I want to post that info to channel.

**Reply by MCausc78** [reply]
use `discord.Webhook`/`discord.SyncWebhook`

**Reply by No767** [reply]
@MCausc78 This doesn't help the situation, nor does it explain what to do.

**Reply by No767** [reply, marked as answer]
@denisyakimov07 You would ideally want to restructure this into an stronger link. From game sever to the bot directly would be better. Trying to get the logs to an API and then back down doesn't make any sense as now you have weaker links between those services and the risk of failures would increase. You want to design your infra to minimize points of failure. So that's why I suggested directly hitting the bot instead. If you do go that route, then I would propose hooking up an aiohttp web server along your bot, and maintaining an websocket connection between your game server and aiohttp web server.

 One of the factors that I would think about here is whether it's worth it to take these logs and whether the logs sent to Discord would be of any use. The logs that are emitted should have some sort of data or weight that either can be used for further analysis or to ensure that an rationale decision can be taken in the future. So sometimes when you think about it, is it *really* worth logging this data?

**Reply by denisyakimov07** [reply]
@No767  you right, thank you for your help. My idea to create message in discord channel and modified massage to keep current player names list. Server has dashboard statistic etc. also API, and webhooks settings. What you would recommend in this case?

**Reply by No767** [reply]
@denisyakimov07 I wouldn't really recommend doing what you referenced above. It would hit the ratelimits very quickly, and can be an security vulnerability by DDOS. For example, I could flood your game server with fake players and cause your bot to edit messages so quickly that it would hit the Discord ratelimit, and if it's sustained, an temp Cloudflare ban. Keeping an list of currently active players and editing it on Discord in theory sounds like a nice feature, but in practice, it would break down quickly. There isn't any good alternative to how you want to track player joins as Discord fundamentally is designed for chatting, not keeping logs or an active player list.

When it comes to the dashboard statistics, I can't figure out any good way to integrate it with Discord unless you write an command that fetches dashboard statistics somehow and parses it out to an image and sends that. In regards to API and webhook settings, not sure what to recommend.
